### PR TITLE
feat: remove Bandwidthmeter type

### LINKED ISSUE #8643: remove Bandwidthmeter type
### Describe the feature

this type isn't really useful because for metrics grafana/prometheus has builtin metrics for bandwidth and we don't even set metrics for these
https://github.com/paradigmxyz/reth/blob/40d8ac099f1bf9eb9640ddf0665e9c7461c0f80a/crates/net/common/src/bandwidth_meter.rs#L53-L53

## TODO
* remove type
* remove usage

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kamuik16**
Closes #8643

### CODE REVIEW

**Comment by kamuik16** [comment]
will mark ready for review after refactoring tests.

**Comment by kamuik16** [comment]
ready for review. cc @mattsse @Rjected @emhane.

File: crates/net/common/src/bandwidth_meter.rs [hunk 1]
```diff
@@ -24,65 +24,16 @@
...
-                inbound: AtomicU64::new(0),
-                outbound: AtomicU64::new(0),
-            }),
-        }
-    }
-}
-
 /// Wraps around a single stream that implements [`AsyncRead`] + [`AsyncWrite`] and meters the
 /// bandwidth through it
 #[derive(Debug)]
```
Comment (mattsse) [inline crates/net/common/src/bandwidth_meter.rs:39]: this type is now just a wrapper+delegate and now effectively useless.

let's remove this file entirely

**Review by mattsse — CHANGES_REQUESTED** [review]
great!

the meteredstream no longer does any metering and we can remove this

File: crates/net/common/src/bandwidth_meter.rs (hunk 1 above)
Comment (kamuik16) [reply crates/net/common/src/bandwidth_meter.rs:39]: > this type is now just a wrapper+delegate and now effectively useless.
> 
> let's remove this file entirely

not to misunderstand, I should completely remove `MeteredStream`?

Comment (mattsse) [reply crates/net/common/src/bandwidth_meter.rs:39]: yep, this type no longer does anything useful

Comment (kamuik16) [reply crates/net/common/src/bandwidth_meter.rs:39]: @mattsse 
```rust
/// The type of the underlying peer network connection.
pub type EthPeerConnection = EthStream<P2PStream<ECIESStream<MeteredStream<TcpStream>>>>;

/// Various connection types that at least support the ETH protocol.
pub type EthSatelliteConnection =
    RlpxSatelliteStream<ECIESStream<MeteredStream<TcpStream>>, EthStream<ProtocolProxy>>;
```
what to do with these types which utilizes `MeteredStream`?

Comment (kamuik16) [reply crates/net/common/src/bandwidth_meter.rs]: I think @mattsse, this is done.

**Comment by kamuik16** [comment]
@mattsse, removed `MeteredStream`. ready for review.

**Review by mattsse — APPROVED** [review]
sweet! tysm

**Comment by kamuik16** [comment]
> sweet! tysm

the merge queue is complaining about merge conflicts, i think it's cargo update pr. wdyt?

**Comment by mattsse** [comment]
resolved conflicts

**Comment by mattsse** [comment]
ah, there's a pr in the queue that introduces conflicts,
will resolve once merged

**Pull request merged**